tests/eosio.msig_tests.cpp

```diff
@@ -1366,4 +1366,170 @@ BOOST_FIXTURE_TEST_CASE( unnecessary_approve_logic, eosio_msig_tester ) try {
+                              ("permission", "perm")
+                              ("parent", "active")
+                              ("auth",  authority{ 1, {}, {permission_level_weight{ {N(sendinline), config::eosio_code_name}, 1}}, {} })
+   );
+   produce_blocks();
+   
+   base_tester::push_action( config::system_account_name, N(linkauth), N(alice), mvo()
+                              ("account", "alice")
+                              ("code", config::system_account_name)
+                              ("type", "reqauth")
+                              ("requirement", "perm")
+   );
+   produce_blocks();
+   
+   transaction_trace_ptr trx_trace;
+   trx_trace = base_tester::push_action( N(sendinline), N(send), N(bob), mvo()
+                                          ("contract", config::system_account_name)
+                                          ("action_name", "reqauth")
+                                          ("auths", std::vector<permission_level>{ {N(alice), N(perm)} })
+                                          ("payload", act.data)
+   );
+   produce_blocks();
+
+   BOOST_REQUIRE( bool(trx_trace) );
+   BOOST_REQUIRE( trx_trace->receipt.valid() );
+   BOOST_REQUIRE_EQUAL( transaction_receipt::executed, trx_trace->receipt->status );
+   BOOST_REQUIRE_EQUAL( 2, trx_trace->action_traces.size() );
+   
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{1}, trx_trace->action_traces.at(0).action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{0}, trx_trace->action_traces.at(0).creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{0}, trx_trace->action_traces.at(0).closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( N(sendinline), action_name{trx_trace->action_traces.at(0).receiver} );
+   BOOST_REQUIRE_EQUAL( N(sendinline), name{trx_trace->action_traces.at(0).act.account} );
+   BOOST_REQUIRE_EQUAL( N(send), name{trx_trace->action_traces.at(0).act.name} );
+   BOOST_REQUIRE_EQUAL( N(bob), name{trx_trace->action_traces.at(0).act.authorization[0].actor} );
+   BOOST_REQUIRE_EQUAL( N(active), name{trx_trace->action_traces.at(0).act.authorization[0].permission} );
+   
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{2}, trx_trace->action_traces.at(1).action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{1}, trx_trace->action_traces.at(1).creator_action_ordinal );
+   BOOST_REQUIRE_EQUAL( fc::unsigned_int{1}, trx_trace->action_traces.at(1).closest_unnotified_ancestor_action_ordinal );
+   BOOST_REQUIRE_EQUAL( N(eosio), action_name{trx_trace->action_traces.at(1).receiver} );
+   BOOST_REQUIRE_EQUAL( N(eosio), name{trx_trace->action_traces.at(1).act.account} );
+   BOOST_REQUIRE_EQUAL( N(reqauth), name{trx_trace->action_traces.at(1).act.name} );
+   BOOST_REQUIRE_EQUAL( N(alice), name{trx_trace->action_traces.at(1).act.authorization[0].actor} );
+   BOOST_REQUIRE_EQUAL( N(perm), name{trx_trace->action_traces.at(1).act.authorization[0].permission} );
+
+   act = get_action(N(eosio.msig), N(approve), {}, mvo()
```

Comment:
Even though the payload is the same for both `approve` and `unapprove` actions in the tests below, you should still split this into a `approve_act` and `unapprove_act` (using the appropriate action to generate the action payload for each) and then use the appropriate `data` field in the checks below.

Reply:
Separated `act` into `approve_act` and `unapprove_act`.